include/revng/Pipes/PipelineManager.h

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <string>
+#include <vector>
+
+#include "llvm/ADT/StringRef.h"
+
+#include "revng/Pipeline/Context.h"
+#include "revng/Pipeline/Runner.h"
+#include "revng/Pipeline/SavableObject.h"
+#include "revng/Pipes/LLVMContextWrapper.h"
+#include "revng/Pipes/ModelGlobal.h"
+
+namespace revng::pipes {
+
+/// This is a "God Object" that can be used when there will be exactly one
+/// Pipeline spawned by this process.
+///
+/// It will care of all configurations including llvm ones and provide a runner
+/// in a usable state.
+class PipelineManager {
```

Comment:
Document all methods.
* What's the difference between `create` and `createFromMemory`? One might think the second option does not use the disk at all.
* What to all the `recalculate*` methods do?
* What's the difference between the *current* state and the *last* state? What are the costs involved in calling `getCurrentState` vs `getLastPipelineState`?

Reply:
Document `writeAllPossibleTargets`, `getTargetsAvailableFor`, `create`, `createFromMemory`.